build-logic/binary-compatibility/src/main/groovy/gradlebuild/binarycompatibility/BinaryCompatibilityHelper.groovy

```diff
@@ -43,24 +43,25 @@ class BinaryCompatibilityHelper {
         String currentVersion
     ) {
         japicmpTask.tap {
+            doNotTrackState("classloading issues with rules")
```

Comment:
Unless there is a disagreement, I am going to merge this after @DPUkyle approves, with this in place. Investigation will continue after this.

Reply:
SGTM. @big-guy and I discussed offline and are fine if this task is not up-to-date/cacheable as it's _mostly_ invoked asynchronously on CI.